thunder/transforms/quantization.py

```diff
@@ -294,3 +296,221 @@ def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilo
+                    new_compute_inputs.append(proxy_lora_b)
+                    new_bsyms.append(
+                        prims.check_tensor_shape_and_metadata.bind(
+                            proxy_lora_a,
+                            qs["lora_a.shape"],
+                            device,
+                            qs["lora_a.dtype"],
+                            False,
+                            output=None,
+                        )
+                    )
+                    new_bsyms.append(
+                        prims.check_tensor_shape_and_metadata.bind(
+                            proxy_lora_b,
+                            qs["lora_b.shape"],
+                            device,
+                            qs["lora_b.dtype"],
+                            False,
+                            output=None,
+                        )
+                    )
+                    # this is not good, because we will have several traces...
+                    additional_proxies[n_lora_a] = proxy_lora_a
+                    additional_proxies[n_lora_b] = proxy_lora_b
+
+        prologue_trace.bound_symbols[-1:-1] = new_bsyms
+        prologue_trace.set_provenance(thunder.core.trace.TraceProvenance("lora linear pass"))
+
+        computation_proxy_map = {
+            csym.name: dict(
+                shape=psym.shape,
+                dtype=psym.dtype,
+            )
+            for psym, csym in zip(prologue_trace.bound_symbols[-1].args[0][0], computation_trace.args)
+            if psym.shape != csym.shape or psym.dtype != csym.dtype
+        }
+
+        new_computation_trace = trace_with_replaced_proxy_metadata(computation_trace, computation_proxy_map)
+        bound_symbols = new_computation_trace.bound_symbols
+        new_computation_trace.bound_symbols = []
+
+        new_computation_trace.args = (*new_computation_trace.args, *new_compute_inputs)
+        new_computation_trace._siginfo.args = [(a.name, None) for a in new_computation_trace.args]
+
+        with tracectx(new_computation_trace):
+            new_bindings = [
+                thunder.core.prims.unpack_trivial.bind(i, output=i, name=i.name) for i in new_compute_inputs
+            ]
+
+        for idx, bsym in enumerate(bound_symbols):
+            if bsym.sym != prims.unpack_trivial:
+                break
+            new_computation_trace.bound_symbols.append(bsym.from_bsym())
+        new_computation_trace.bound_symbols += new_bindings
+
+        swap_map = dict()
+
+        for bsym in bound_symbols[idx:]:
+            if bsym.sym == thunder.torch.linear and bsym.args[1].name in lora_linear_proxies:
+                n = lora_linear_proxies[bsym.args[1].name]
+                with tracectx(new_computation_trace):
+                    new_computation_trace.push_scope(new_computation_trace.bound_symbols)
+                    lora_a_bsym = thunder.torch.transpose(additional_proxies[f"{n}.lora_a"], 1, 0)
```

Comment:
Could we rename the variable? `lora_a_t` or somesuch, it is not a bsym but a proxy. (Similar below).

Reply:
Sure thing!